tests/test_config.py

```diff
@@ -633,3 +635,48 @@ class Child(Parent):
         model_config: ConfigDict = {'str_to_lower': True}
 
     assert Child.model_config == {'extra': 'allow', 'str_to_lower': True}
+
+
+def test_replace_types():
+    LaxStr = Annotated[str, BeforeValidator(str_validator)]
+
+    class LaxStrBase(BaseModel):
+        model_config: ConfigDict = {'replace_types': {str: LaxStr}}
+
+    class Data(LaxStrBase):
+        f: str
+        i: str
+        d: str
+
+    user = Data(f=1.1, i=1, d=Decimal('1.1'))
+
+    assert user.model_dump() == {'f': '1.1', 'i': '1', 'd': '1.1'}
```

Comment:
Consider parameterizing this test case – I would expect to see test examples for Nonetypes – ex, `None, 0, 'nan'`.